src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -173,6 +178,38 @@ public List<Product> findAllProducts() {
         }
     }
 
+    @Override
+    public void addProductToRecentlyViewed(HttpServletRequest httpServletRequest, Product product) {
+        var currentSession = httpServletRequest.getSession();
+        LinkedList<Product> productsFromSession = (LinkedList<Product>) currentSession.getAttribute(RECENTLY_VIEWED_PRODUCTS_SESSION_ATTRIBUTE);
+        LinkedList<Product> products;
+        if (productsFromSession == null) {
+            products = new LinkedList<>();
+            products.addFirst(product);
+        } else {
+            products = productsFromSession;
+            var firstProduct = products.getFirst();
+            if (!Objects.equals(firstProduct.getId(), product.getId())) {
+                products.addFirst(product);
+            }
+            if (products.size() > 3) {
```

Comment:
Move recently view products list size to class constant

Reply:
Ok